### DISCUSSION TITLE
Depreciation Warning from asyncio in Command and OProc _init

### DISCUSSION THREAD

**Discussion by Hos73** [amoffat/sh#682] [Q&A] [answered]
sh package version: 2.0.4
I was checking for Depreciations in my code and stumbled across depreciations in the mentioned _inits for Command. In particular line 665 and 2398
`sh.py:665: DeprecationWarning: There is no current event loop asyncio.get_event_loop()`
`sh.py:2398: DeprecationWarning: There is no current event loop loop = asyncio.get_event_loop()`

Depreciation note from asyncio
`Note In Python versions 3.10.0–3.10.8 and 3.11.0 this function (and other functions which use it implicitly) emitted a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) if there was no running event loop, even if the current loop was set on the policy. In Python versions 3.10.9, 3.11.1 and 3.12 they emit a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) if there is no running event loop and no current loop is set. In some future Python release this will become an error`

Am I using command incorrectly by not creating an asyncio event loop? Or is this just unintended depreciations? Any assistance would be great, trying to address this as it is showing up in my pytest code outputs, thank you!

**Comment by amoffat** [comment, marked as answer]
You're fine, this looks like an issue on sh's part. I've made an issue #683 to fix this. Thank you for reporting.

**Reply by Hos73** [reply]
Thank you very much!
